### PR TITLE
feat: add support for the Web Share Target API

### LINKED ISSUE #1093: Web Share Target for installed PWA
### Clear and concise description of the problem

https://discord.com/channels/1044887051155292200/1046769315405365288/1063504082247102524

Hi everyone, new Elk user here, congrats on the app! Are there currently plans/is anyone working on using the Web Share Target API for installed PWAs? (https://developer.chrome.com/articles/web-share-target/)

### Suggested solution

https://developer.chrome.com/articles/web-share-target/

### Alternative

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by horvbalint**
Hi everyone,

I noticed that there is currently no support for the Web Share Target API in Elk, so I implemented basic support for it :)
I talked with @userquin on Discord and he opened this issue in the repo: [https://github.com/elk-zone/elk/issues/1093](https://github.com/elk-zone/elk/issues/1093)

Web Share Target API would allow users (on supported platforms) who have Elk installed as a PWA, to choose some media (image/video) and choose Elk in the OS's share-with dialog. This would open Elk and automatically upload the media and focus the compose section of the home page. If not some media but some text or an URL was shared, then again the compose textarea is focused and the text is inserted inside it.

I hope that my changes are OK as the codebase and TypeScript is pretty new for me. I had to extend the ManifestOptions interface of 'vite-plugin-pwa', because it didn't know about the share_target field of the WebApp Manifest and I am not sure if this is the right thing to do.

Here is an example on how the share screen would look like:
Screenshot_2023-01-13-21-55-29-199_com miui gallery

closes #1093

### CODE REVIEW

**Comment by userquin** [comment]
The PWA must be installed and the user must be logged in: the `home` page requires auth.

Right now we cannot use `indexedDB` on `fetch` listener, I don't know why it is failing.

The solution: redirect to home page with `?share-target=true`, then in oauth middleware check if we are being redirected from share target, if so just send to `share-target` page.

Next step will be to include also on `share-page` a preview then listen to auth and then redirect to `home` page with the same logic.

**Review by userquin — APPROVED** [review]

File: components/publish/PublishWidget.vue [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <script setup lang="ts">
 import { EditorContent } from '@tiptap/vue-3'
 import type { mastodon } from 'masto'
-import type { Ref } from 'vue'
 import type { Draft } from '~/types'
+import { useWebShareTarget } from '~/composables/web-share-target'
```
Comment (danielroe) [inline components/publish/PublishWidget.vue:5]: We can omit this as it will be auto-imported.

```suggestion
```

File: locales/en-GB.json [hunk 2]
```diff
@@ -282,6 +282,11 @@
       "label": "Logged in users"
     }
   },
+  "share-target": {
+    "description": "Elk can be configured so that you can share content from other applications, simply install Elk on your device or computer and sign in.",
+    "hint": "In order to share content with Elk, Elk must be installed and you must be logged.",
```
Comment (danielroe) [inline locales/en-GB.json:287]:
```suggestion
    "hint": "In order to share content with Elk, Elk must be installed and you must be signed in.",
```

File: locales/en-US.json [hunk 3]
```diff
@@ -359,6 +359,11 @@
       "label": "Wellness"
     }
   },
+  "share-target": {
+    "description": "Elk can be configured so that you can share content from other applications, simply install Elk on your device or computer and sign in.",
+    "hint": "In order to share content with Elk, Elk must be installed and you must be logged.",
```
Comment (danielroe) [inline locales/en-US.json:364]:
```suggestion
    "hint": "In order to share content with Elk, Elk must be installed and you must be signed in.",
```

File: pages/share-target.vue [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<script setup>
+import { useWebShareTarget } from '~/composables/web-share-target'
+
```
Comment (danielroe) [inline pages/share-target.vue:3]:
```suggestion
```

**Review by danielroe — APPROVED** [review]
Beautiful work.

**Pull request merged**